### ISSUE TITLE
how to get good mesh?

### ISSUE THREAD

**Issue by hanjoonwon** [autonomousvision/sdfstudio#293]
meshoutput
![image](https://github.com/autonomousvision/sdfstudio/assets/125524043/51bd4412-ea0a-4c5e-9b06-7941f350e034)

my image like this 
![image](https://github.com/autonomousvision/sdfstudio/assets/125524043/1fa3d129-4ce2-427d-9d2b-7af1e23a6a76)

I went through the following process

1. ns-process-data images --data data/owl --output-dir process/owl    

2.python  scripts/datasets/process_nerfstudio_to_sdfstudio.py  --mono-prior --data-type colmap 
--scene-type object --data process/owl --output-dir sdfdata/owl  
--omnidata-path omnidata/omnidata_tools/torch --pretrained-models 
omnidata/omnidata_tools/torch/pretrained_models

3 .ns-train neus-facto --pipeline.datamanager.train-num-rays-per-batch 2048 
--pipeline.model.sdf-field.geometric-init True  --pipeline.model.sdf-field.use-grid-feature True
 --pipeline.model.sdf-field.inside-outside False --pipeline.model.background-model mlp 
 --pipeline.model.mono-depth-loss-mult 0.01 --pipeline.model.mono-normal-loss-mult 0.01
  --pipeline.model.sdf-field.bias 0.3   --vis wandb  --trainer.steps_per_save 5000  
  --trainer.steps-per-eval-image 5000  --trainer.max-num-iterations 300000
  --experiment-name neusfactoowl sdfstudio-data --data sdfdata/owl
  
  I've run neus-facto training several times with different parameters, but I'm still not getting a good mesh.
It works fine in nerfstudio and sugar, so what could be the reason?

**Comment by niujinshuchong** [comment]
Hi, Could you try to disable using grid-features with `--pipeline.model.sdf-field.use-grid-feature False` since the hash grids is very sensitive.

**Comment by hanjoonwon** [comment]
> Hi, Could you try to disable using grid-features with `--pipeline.model.sdf-field.use-grid-feature False` since the hash grids is very sensitive.

@niujinshuchong  also i got poor result, my parameters are wrong??

**Comment by niujinshuchong** [comment]
Oh, Could you also try to disable the mono-depth loss? The mono-depth loss uses scale-invariant loss and this needs to compute the alignment between rendered depth map and monocular depth map. Therefore, we should sample the training rays from the same image. The default setting of neus-facto will sample rays across all training images and this will likely introduce issues during optimisation.

**Comment by hanjoonwon** [comment]
> Oh, Could you also try to disable the mono-depth loss? The mono-depth loss uses scale-invariant loss and this needs to compute the alignment between rendered depth map and monocular depth map. Therefore, we should sample the training rays from the same image. The default setting of neus-facto will sample rays across all training images and this will likely introduce issues during optimisation.

@niujinshuchong  
ns-train neus-facto
 --pipeline.model.sdf-field.use-grid-feature False --pipeline.model.background-model mlp
  --pipeline.model.mono-depth-loss-mult 0.0 --pipeline.model.mono-normal-loss-mult 0.0
 --pipeline.model.sdf-field.inside-outside False
 
I don't know if I'm understanding you correctly, but is this what you're saying? the results were also not good

**Comment by hanjoonwon** [comment]
> Oh, Could you also try to disable the mono-depth loss? The mono-depth loss uses scale-invariant loss and this needs to compute the alignment between rendered depth map and monocular depth map. Therefore, we should sample the training rays from the same image. The default setting of neus-facto will sample rays across all training images and this will likely introduce issues during optimisation.

@niujinshuchong  
![image](https://github.com/autonomousvision/sdfstudio/assets/125524043/6bb81047-ad72-43cf-a844-f791577d178e)

I trained with 
ns-train neus-facto --pipeline.datamanager.train-num-rays-per-batch 2048 \
--pipeline.model.sdf-field.use-grid-feature False \
--pipeline.model.sdf-field.inside-outside False \
--pipeline.model.background-model mlp \
--pipeline.model.mono-depth-loss-mult 0.0 \
--pipeline.model.mono-normal-loss-mult 0.01 \
--vis wandb \
--trainer.steps_per_save 5000 \
--trainer.steps-per-eval-image 5000 \
--trainer.max-num-iterations 300000 \
--experiment-name omnineusowl sdfstudio-data


but got weired result..
my images are here https://drive.google.com/drive/folders/1lHsiW8MGQcVTrQT9LtJjH8Z6ChMiiVlf?usp=drive_link
and wandb log https://wandb.ai/ju805604/sdfstudio/runs/sqw2b32u?workspace=user-ju805604
